### PR TITLE
feat(js-loops): add challenges

### PR DESCRIPTION
**Pull request by F-Kirchhoff**
rendered: https://github.com/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops

Links to CSB:
- [for-console](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/for-console?file=/README.md)
- [for-stars](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/for-stars?file=/README.md)
- [forin-select-nav](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/forin-select-nav?file=/README.md)
- [forof-list](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/forof-list?file=/README.md)
- [add pixels](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/pixels?file=/README.md)
- [while-random-number](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/while-random-number?file=/README.md)

refers to
- neuefische/web-curriculum-new-format#4

### CODE REVIEW

**Review by PReichetanz — DISMISSED** [review]
Really nice challenge!

[Please find my solution here](https://codesandbox.io/s/js-loops-add-pixels-9lzn7x?file=/js/index.js).

File: sessions/js-loops/for-stars/js/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const main = document.querySelector('[data-js="main"]');
+
+// Part 1: Create 5 stars using a for loop
+
+// --v-- write/change code here --v--
+
+// --^-- write/change code here --^--
+
+// Part 2: for loop that manipulate the src property of the created stars (based on an input)
```
Comment (CJAkkuec) [inline sessions/js-loops/for-stars/js/index.js:17]:
```suggestion
// Part 2: for loop that manipulates the src property of the created stars (based on an input)
```

File: sessions/js-loops/while-random-number/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+console.clear();
+
+// `while` loop
+
+const number = 0;
```
Comment (CJAkkuec) [inline sessions/js-loops/while-random-number/index.js:5]:
```suggestion
let number = 0;
```

File: sessions/js-loops/while-random-number/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+const count = 0;
```
Comment (CJAkkuec) [inline sessions/js-loops/while-random-number/index.js:6]:
```suggestion
let count = 0;
```

**Review by CJAkkuec — COMMENTED** [review]
Partial review, tbc.

**Comment by CJAkkuec** [comment]
**Review (WIP)**
> [for-console](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/for-console?file=/README.md)
LGTM! 🚀 
[My solution.](https://codesandbox.io/s/bold-hertz-937zvn?file=/index.js)

>[for-stars](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/for-stars?file=/README.md)
In progress ...

>[forin-select-nav](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/forin-select-nav?file=/README.md)
-

>[forof-list](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/forof-list?file=/README.md)
-

>[add pixels](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/pixels?file=/README.md)
Reviewed by @PReichetanz 

>[while-random-number](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-loops/sessions/js-loops/while-random-number?file=/README.md)
For me, this only worked when replacing `const` with `let` while declaring `number` and `count`.
[My solution.](https://codesandbox.io/s/affectionate-microservice-wngkkc?file=/index.js)

**Comment by Kheirah** [comment]
Solutions:

[for-console](https://codesandbox.io/s/crazy-newton-zd8gy9?file=/index.js)
[for-stars](https://codesandbox.io/s/affectionate-williams-9oqjns?file=/js/index.js)
[forin-select-nav](https://codesandbox.io/s/gallant-volhard-hgv75z?file=/js/index.js)
[forof-list](https://codesandbox.io/s/adoring-maria-2gwts1?file=/js/index.js)
[add pixels](https://codesandbox.io/s/confident-aryabhata-znqdw2?file=/js/index.js)
[while-random-number](https://codesandbox.io/s/admiring-hoover-vxfcsy?file=/index.js)

File: sessions/js-loops/for-stars/js/index.js (hunk 1 above)
Comment (PReichetanz) [reply sessions/js-loops/for-stars/js/index.js:17]: I rephrased the headline according to the headline in the readme file.

File: sessions/js-loops/for-console/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+To be clear: Loops can be frightening at the beginning. But don't worry, we are going to make you familiar with the syntax and it will be fine ;)
+
+## Task
+
+Log the numbers from 0 to 100 (including) into the console. Use a for loop.
+
+Make sure your for loop has
+
+- an internal counter variable
+- a criteria to be met by the counter variable
```
Comment (Kheirah) [inline sessions/js-loops/for-console/README.md:12]:
```suggestion
- a criterion to be met by the counter variable
```

File: sessions/js-loops/for-stars/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+### Part 1 : Creating Stars
+
+Create five stars in the browser window using a for loop.
+
+The following hints may guide you:
+
+- The loop should have
+  - an internal counter variable
+  - a criteria to be met by the counter variable
```
Comment (Kheirah) [inline sessions/js-loops/for-stars/README.md:17]:
```suggestion
  - a criterion to be met by the counter variable
```

File: sessions/js-loops/for-stars/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+#### Part 2.1 Refactor Part 1
+
+This code can only work if there are any HTML elements with a data-js attribute and a value "star". This is why we need to refactor part 1:
+
+Add `star.dataset.js = 'star'` to each created element. Make sure to do this before you append it to the parent element.
+
+#### Part 2.2 Add Rating
+
+No we can write the `fillStars` function:
```
Comment (Kheirah) [inline sessions/js-loops/for-stars/README.md:37]:
```suggestion
Now we can write the `fillStars` function:
```

File: sessions/js-loops/for-stars/js/index.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+const stars = document.querySelectorAll('[data-js="star"]');
+stars.forEach((star, index) => {
+	star.addEventListener('click', () => {
+		fillStars(index);
+	});
+});
```
Comment (Kheirah) [inline sessions/js-loops/for-stars/js/index.js:24]:
```suggestion
for (let i = 0; i < stars.length; i++) {
  stars[i].addEventListener('click', () => {
    fillStars(i);
  });
```

Man könnte überlegen anstelle der forEach-Schleife die for-Schleife zu nutzen. Hier könnte man sich daran richten, ob die forEach-Schleife schon bekannt ist. Wenn nicht, hinterlasse ich einen Code Vorschlag für die for-Schleife.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Ein paar kleine sprachliche Anpassungen und ein Code Vorschlag eine forSchleife anstelle einer forEach-Schleife zu nutzen.

**Review by Kheirah — APPROVED** [review]
Yay, loops! Round and round they go

**Pull request merged**